src/OpenTelemetry/Logs/OpenTelemetryLogger.cs

```diff
@@ -52,6 +52,7 @@ public void Log<TState>(LogLevel logLevel, EventId eventId, TState state, Except
             this.provider.Processor?.OnEnd(record);
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```

Comment:
Also thinking if we should remove this check completely! If we remove it, then `logger.Log(LogLevel.None, "I dont want to see this")` will appear in logs. :-/